### PR TITLE
Time-format HH:MM:SS

### PR DESCRIPTION
**Pull request by germanysources**
This pull-request added parsing for time-format HH:MM:SS and for time-format HHMMSS.
The huge difference in the testclass comes from the pretty-printer.

### CODE REVIEW

File: src/zcl_text2tab_parser.clas.testclasses.abap [hunk 1]
```diff
@@ -374,36 +385,69 @@ class ltcl_text2tab_parser_test implementation.
...
+
+    cut = zcl_text2tab_parser=>create( l_exp_result ).
+
+    " not successfull parsing
+    try.
+        cut->parse( exporting i_data = l_with_invalid_time
+          importing e_container = l_act_result ).
+        cl_abap_unit_assert=>fail( msg = 'no exception when given invalid time' ).
+      ##no_handler
+      catch zcx_text2tab_error.
```
Comment (sbcgua) [inline src/zcl_text2tab_parser.clas.testclasses.abap]: please add the check of the exception code (that it was exactly the expected exception)

File: src/zcl_text2tab_parser.clas.testclasses.abap [hunk 2]
```diff
@@ -374,36 +385,69 @@ class ltcl_text2tab_parser_test implementation.
...
+    endtry.
+
+    " successfull parsing
+    cut->parse( exporting i_data = l_with_valid_time
+      importing e_container = l_act_result ).
```
Comment (sbcgua) [inline src/zcl_text2tab_parser.clas.testclasses.abap]: please indent like
```
  cut->parse(
    exporting
      i_data = l_with_valid_time
    importing
      e_container = l_act_result )
```

File: src/zcl_text2tab_parser.clas.testclasses.abap [hunk 3]
```diff
@@ -374,36 +385,69 @@ class ltcl_text2tab_parser_test implementation.
...
+
+    cl_abap_unit_assert=>assert_equals( exp = l_exp_result
+      act = l_act_result msg = 'parsing should be sucessfull with correct time' ).
```
Comment (sbcgua) [inline src/zcl_text2tab_parser.clas.testclasses.abap]: and similarly here
```
  exp = 
  ext = 
  msg = 
```

File: src/zcl_text2tab_parser.clas.testclasses.abap [hunk 4]
```diff
@@ -465,172 +509,172 @@ class ltcl_text2tab_parser_test implementation.
...
-    test_parse_positive TCHAR    'ABC'             'ABC'.
-    test_parse_positive TSTRING  'The string test' 'The string test'.
-    test_parse_positive TALPHA   '100000'          '0000100000'.
-    test_parse_positive TNUMBER  '2015'            '2015'.
-    test_parse_positive TINTEGER '123'             123.
-    test_parse_positive TRAW     '8E'              '8E'.
-    test_parse_positive TFLOAT   '1.123456789'     '1.123456789'.
-    test_parse_positive TFLOAT   '"1.123456789"'   '1.123456789'. " Quoted data, issue#6
-    test_parse_positive TNUMBER  '"2015"'          '2015'.        " Quoted
+    test_parse_positive tdate    '01.02.2015'      '20150201'.
```
Comment (sbcgua) [inline src/zcl_text2tab_parser.clas.testclasses.abap]: hmmm, does this still pass the unit test ?
```abap
test_parse.
  ...
  read table lt_components into ls_component with key name = '&1' " << I'd expect this fails with lower case ...
```

**Comment by sbcgua** [comment]
Also please have a look at abaplint check after the commit

**Comment by germanysources** [comment]
I fixed the pretty-printer back and changed the intention.

**Comment by sbcgua** [comment]
Looks OK, just fix the error code in the parse time - `code = 'IT'` if I understood well.
image

**Comment by germanysources** [comment]
Error code is added

**Comment by sbcgua** [comment]
Merging then. Thanks for the contribution ! :)

**Pull request merged**